src/extension/wps/src/main/java/au/org/emii/aggregator/variable/UnpackerOverrides.java

```diff
@@ -0,0 +1,114 @@
+package au.org.emii.aggregator.variable;
+
+import ucar.ma2.DataType;
+
+import java.util.Arrays;
+
+/**
+ * Optional overrides for values to be used in the unpacked variable
+ */
+public class UnpackerOverrides {
+    private final Number newFillerValue;
+    private final Number[] newValidRange;
+    private final Number newValidMin;
+    private final Number newValidMax;
+    private final Number[] newMissingValues;
+    private final DataType newDataType;
+
+    private UnpackerOverrides(Builder builder) {
+        this.newFillerValue = builder.newFillerValue;
+        this.newValidRange = builder.newValidRange == null ? null :
+            Arrays.copyOf(builder.newValidRange, builder.newValidRange.length);
+        this.newValidMin = builder.newValidMin;
+        this.newValidMax = builder.newValidMax;
+        this.newMissingValues = builder.newMissingValues == null ? null :
+            Arrays.copyOf(builder.newMissingValues, builder.newMissingValues.length);
+        this.newDataType = builder.newDataType;
+    }
+
+    public Number getNewFillerValue() {
```

Comment:
I benchmarked using primitive types/autoboxing and there was not enough of a difference to justify all the extra code I'd have to write (I couldn't see any difference actually)